File: corelib/src/main/java/org/apache/cordova/fragment/BaseFragment.java
package org.apache.cordova.fragment;

import android.content.Context;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.ViewModel;
import androidx.fragment.app.Fragment;

import org.apache.cordova.R;


public abstract class BaseFragment extends Fragment {

    protected Callback mCallback;

    @Override
    public void onResume() {
        super.onResume();
//        if (mCallback != null) {
//            //mCallback.hideTabLayout();
//        }
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (context instanceof Callback) {
            this.mCallback = (Callback) context;
        } else {
            throw new RuntimeException(context + " must implement Callback");
        }
    }

    //Callback
    public interface Callback {

        void onClick(ViewModel navItem);

        void showToolbar(boolean visible);

        void replaceFragment(Fragment preferencesFragment);
    }
}

File: kwk/corelib/src/main/java/org/apache/cordova/fragment/BaseFragment.kt
package org.apache.cordova.fragment

import android.content.Context
import androidx.appcompat.ViewModel
import androidx.fragment.app.Fragment

abstract class BaseFragment : Fragment() {
    protected var mCallback: Callback? = null

    override fun onResume() {
        super.onResume()
        //        if (mCallback != null) {
//            //mCallback.hideTabLayout();
//        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is Callback) {
            this.mCallback = context
        } else {
            throw RuntimeException("$context must implement Callback")
        }
    }

    //Callback
    interface Callback {
        fun onClick(navItem: ViewModel?)

        fun showToolbar(visible: Boolean)

        fun replaceFragment(preferencesFragment: Fragment?)
    }
}
